--- config.def.h	2026-01-16 19:23:34.908060745 +0100
+++ config.def.h	2026-01-16 19:23:05.388599106 +0100
@@ -96,32 +96,32 @@
 /* Terminal colors (16 first used in escape sequence) */
 static const char *colorname[] = {
 	/* 8 normal colors */
-	"black",
-	"red3",
-	"green3",
-	"yellow3",
-	"blue2",
-	"magenta3",
-	"cyan3",
-	"gray90",
+	"#414868",
+	"#f7768e",
+	"#73daca",
+	"#ff9e64",
+	"#7aa2f7",
+	"#bb9af7",
+	"#7dcfff",
+	"#c0caf5",

 	/* 8 bright colors */
 	"gray50",
-	"red",
-	"green",
-	"yellow",
-	"#5c5cff",
-	"magenta",
-	"cyan",
-	"white",
+	"#f7768e",
+	"#73daca",
+	"#e0af68",
+	"#7aa2f7",
+	"#bb9af7",
+	"#7dcfff",
+	"#c0caf5",

 	[255] = 0,

 	/* more colors can be added after 255 to use with DefaultXX */
-	"#cccccc",
-	"#555555",
-	"gray90", /* default foreground colour */
-	"black", /* default background colour */
+	"#c0caf5",
+	"#1a1b26",
+	"#a9b1d6", /* default foreground colour */
+	"#1a1b26", /* default background colour */
 };


