--- config.def.h	2026-01-18 22:12:24.285934286 +0100
+++ config.def.h	2026-01-18 22:13:01.306436385 +0100
@@ -228,7 +228,34 @@
  * If you want keys other than the X11 function keys (0xFD00 - 0xFFFF)
  * to be mapped below, add them to this array.
  */
-static KeySym mappedkeys[] = { -1 };
+static KeySym mappedkeys[] = {
+  XK_A,
+  XK_B,
+  XK_C,
+  XK_D,
+  XK_E,
+  XK_F,
+  XK_G,
+  XK_H,
+  XK_I,
+  XK_J,
+  XK_K,
+  XK_L,
+  XK_M,
+  XK_N,
+  XK_O,
+  XK_P,
+  XK_R,
+  XK_S,
+  XK_T,
+  XK_U,
+  XK_V,
+  XK_Z,
+  XK_Y,
+  XK_X,
+  XK_Q,
+  XK_W,
+};

 /*
  * State bits to ignore when matching key or button events.  By default,
@@ -350,6 +377,7 @@
 	{ XK_Delete,        XK_ANY_MOD,     "\033[3~",      +1,    0},
 	{ XK_BackSpace,     XK_NO_MOD,      "\177",          0,    0},
 	{ XK_BackSpace,     Mod1Mask,       "\033\177",      0,    0},
+	{ XK_BackSpace,     ControlMask,    "\033\177",      0,    0},
 	{ XK_Home,          ShiftMask,      "\033[2J",       0,   -1},
 	{ XK_Home,          ShiftMask,      "\033[1;2H",     0,   +1},
 	{ XK_Home,          XK_ANY_MOD,     "\033[H",        0,   -1},
@@ -365,6 +392,34 @@
 	{ XK_Next,          ControlMask,    "\033[6;5~",     0,    0},
 	{ XK_Next,          ShiftMask,      "\033[6;2~",     0,    0},
 	{ XK_Next,          XK_ANY_MOD,     "\033[6~",       0,    0},
+  { XK_A,      ControlMask|ShiftMask, "\033[65;6u",    0,    0},
+  { XK_B,      ControlMask|ShiftMask, "\033[66;6u",    0,    0},
+  { XK_C,      ControlMask|ShiftMask, "\033[67;6u",    0,    0},
+  { XK_D,      ControlMask|ShiftMask, "\033[68;6u",    0,    0},
+  { XK_E,      ControlMask|ShiftMask, "\033[69;6u",    0,    0},
+  { XK_F,      ControlMask|ShiftMask, "\033[70;6u",    0,    0},
+  { XK_G,      ControlMask|ShiftMask, "\033[71;6u",    0,    0},
+  // { XK_H,      ControlMask|ShiftMask, "\033[72;6u",    0,    0},
+  { XK_I,      ControlMask|ShiftMask, "\033[73;6u",    0,    0},
+  { XK_J,      ControlMask|ShiftMask, "\033[74;6u",    0,    0},
+  { XK_K,      ControlMask|ShiftMask, "\033[75;6u",    0,    0},
+  // { XK_L,      ControlMask|ShiftMask, "\033[76;6u",    0,    0},
+  { XK_M,      ControlMask|ShiftMask, "\033[77;6u",    0,    0},
+  { XK_N,      ControlMask|ShiftMask, "\033[78;6u",    0,    0},
+  { XK_O,      ControlMask|ShiftMask, "\033[79;6u",    0,    0},
+  { XK_P,      ControlMask|ShiftMask, "\033[80;6u",    0,    0},
+  { XK_Q,      ControlMask|ShiftMask, "\033[81;6u",    0,    0},
+  { XK_R,      ControlMask|ShiftMask, "\033[82;6u",    0,    0},
+  { XK_S,      ControlMask|ShiftMask, "\033[83;6u",    0,    0},
+  { XK_T,      ControlMask|ShiftMask, "\033[84;6u",    0,    0},
+  { XK_U,      ControlMask|ShiftMask, "\033[85;6u",    0,    0},
+  { XK_V,      ControlMask|ShiftMask, "\033[86;6u",    0,    0},
+  { XK_W,      ControlMask|ShiftMask, "\033[87;6u",    0,    0},
+  { XK_X,      ControlMask|ShiftMask, "\033[88;6u",    0,    0},
+  { XK_Y,      ControlMask|ShiftMask, "\033[89;6u",    0,    0},
+  { XK_Z,      ControlMask|ShiftMask, "\033[90;6u",    0,    0},
+  { XK_H,      ControlMask|ShiftMask, "ħ",             0,    0},
+  { XK_L,      ControlMask|ShiftMask, "ł",             0,    0},
 	{ XK_F1,            XK_NO_MOD,      "\033OP" ,       0,    0},
 	{ XK_F1, /* F13 */  ShiftMask,      "\033[1;2P",     0,    0},
 	{ XK_F1, /* F25 */  ControlMask,    "\033[1;5P",     0,    0},

--- x.c	2026-01-19 08:49:59.829017667 +0100
+++ x.c	2026-01-18 22:32:57.884182689 +0100
@@ -1899,10 +1884,16 @@
 		if (mappedkeys[i] == k)
 			break;
 	}
-	if (i == LEN(mappedkeys)) {
-		if ((k & 0xFFFF) < 0xFD00)
-			return NULL;
-	}
+
+  /* This block of code interferes with Ctrl+shift+<Key> combination. */
+  /* The terminal does not have a special string for the Ctrl+shift+<Key> combination. */
+  /* So if i want to make it work, i have to modify the code in config.h key[] array. */
+  /* I have XK_H and XK_L with the Ctrl Shift masks. And the output ħ and ł. */
+  /* This way i can use this keys in tmux. */
+	/* if (i == LEN(mappedkeys)) { */
+	/* 	if ((k & 0xFFFF) < 0xFD00) */
+	/* 		return NULL; */
+	/* } */

 	for (kp = key; kp < key + LEN(key); kp++) {
 		if (kp->k != k)
